cogs/poll.py

```diff
@@ -0,0 +1,131 @@
+import asyncio
+import discord
+from discord.ext import commands
+from cogs.base import BaseCog
+
+
+class Poll(BaseCog):
+
+    @commands.command(pass_context=True)
+    async def poll(self, ctx, question, minutes, *options: str):
+        """ Generates a poll. Consists of a question, the options to choose from and the number of minutes to keep
+        the poll going
+
+        :param ctx: Context of the command message
+        :param question: Question for the poll
+        :param options: List of options to choose from
+        :param minutes: Number of minutes to run the poll for
+        :return: None
+        """
+        if number_of_options_does_not_fall_in_range(options):
+            await ctx.send('Number of options needs to be more than 1 and less than 10. Number of options given: {}.'
+                           .format(len(options)))
+            return
+        await ctx.message.delete()
+        reactions = generate_reactions(options)
+        description = generate_poll_description(options, reactions)
+        description_as_string = description_to_string(description)
+        embed = discord.Embed(title=question, description=description_as_string, colour=discord.Color.orange())
+        embed.set_footer(text='Poll:')
+        poll_message = await ctx.send(embed=embed)
+        await add_reactions_to_the_poll_message(poll_message, reactions, len(options))
+        await asyncio.sleep(int(minutes) * 60)
+        await ctx.send(embed=await generate_results(ctx, poll_message))
+
+
+def number_of_options_does_not_fall_in_range(options) -> bool:
```

Comment:
This does not seem be used more than once. This logic should be pulled out into line 20